app/controllers/stripe_event/webhook_controller.rb

```diff
@@ -1,5 +1,12 @@
 module StripeEvent
   class WebhookController < ActionController::Base
+    before_filter do
+      StripeEvent.authentication_secret.nil? or \
+        authenticate_or_request_with_http_basic do |username, password|
+          password == StripeEvent.authentication_secret
+        end
```

Comment:
Overall, decent looking pull-request, but could we move this portion to more idiomatic ruby?

Either:

```ruby
authenticate_or_request_with_http_basic do |username, password|
  password == StripeEvent.authentication_secret
end if StripeEvent.authentication_secret
```

OR 

```ruby
if StripeEvent.authentication_secret
  authenticate_or_request_with_http_basic do |username, password|
    password == StripeEvent.authentication_secret
  end
end
```